### PR TITLE
refactor: added utility and spacing classes and refactored

### PR DESCRIPTION
**Pull request by TejasShekar**

### CODE REVIEW

File: Components/Buttons/button.css [hunk 1]
```diff
@@ -1,27 +1,33 @@
 .btn {
   background-color: var(--secondary);
-  cursor: pointer;
-  padding: 0.5rem 1rem;
-  font-size: 1.5rem;
+  font-size: 1.3rem;
   border: 1px solid transparent;
   border-radius: 0.475rem;
```
Comment (Maianki) [inline Components/Buttons/button.css:5]: Any specific reason for using such a specific value?

File: Stylesheets/reset.css [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ul {
+  list-style-type: none;
+}
+
+button {
+  cursor: pointer;
+  text-align: center;
+  outline: none;
+  border: none;
+  /* background-color: white; */
```
Comment (ravikumar0403) [inline Stylesheets/reset.css]: remove this commented code

**Review by ravikumar0403 — APPROVED** [review]
Looks good to me

File: Components/Input/input.css [hunk 3]
```diff
@@ -21,19 +20,15 @@ input[type="password"] {
...
+  border: 2px solid var(--success);
 }
 
 #error-input {
-  border: 3px solid var(--danger);
-}
-
-.gap-0 {
-  gap: 0;
+  border: 2px solid var(--danger);
```
Comment (Maianki) [inline Components/Input/input.css]: Suggestion: Avoid using Id selectors until it's really necessary as they have very high specificity.

File: Components/Input/input.css [hunk 4]
```diff
@@ -1,6 +1,5 @@
 .input-container {
   width: 300px;
```
Comment (MaruthiPothuganti) [inline Components/Input/input.css]: This will hinder responsiveness. You can use REMs instead

**Review by darshandamre — APPROVED** [review]
good job Tejas

**Review by Maianki — APPROVED** [review]
Looks good just a small suggestion try raising small PR .

File: Components/Input/input.css (hunk 3 above)
Comment (TejasShekar) [reply Components/Input/input.css]: Will add a proper class and fix it.

File: Components/Buttons/button.css (hunk 1 above)
Comment (TejasShekar) [reply Components/Buttons/button.css:5]: default font size of button text is too small to read. I felt 1.3rem is a sweet spot and has good readbility. So, hardcoded the value.

File: Stylesheets/docs.css [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  color: var(--text-color);
+  text-decoration: underline var(--primary-dark);
+  font-weight: bold;
+}
+
+.component-name {
+  color: var(--primary-dark);
+}
+
+span {
```
Comment (MaruthiPothuganti) [inline Stylesheets/docs.css]: Avoid styling with tag names. Use a class for span and apply  styling to that.

File: Stylesheets/reset.css (hunk 2 above)
Comment (TejasShekar) [reply Stylesheets/reset.css]: done

File: Stylesheets/docs.css [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+.code-block {
+  max-width: 80%;
+  margin: 0 auto;
+}
+
+.toggle {
+  font-size: 1.1rem;
+}
+
+aside {
```
Comment (MaruthiPothuganti) [inline Stylesheets/docs.css]: Same as above. Avoid styling with tag names

File: Stylesheets/reset.css [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+ul {
+  list-style-type: none;
+}
+
+button {
+  cursor: pointer;
+  text-align: center;
+  outline: none;
```
Comment (MaruthiPothuganti) [inline Stylesheets/reset.css]: Its not a good practice to make Outline of a button as none as it. It provides visual feedback for links that have "focus" when navigating a web document using the TAB key (or equivalent). This is especially useful for folks who can't use a mouse or have a visual impairment. If you remove the outline you are making your site inaccessible for these people

**Review by MaruthiPothuganti — COMMENTED** [review]
Suggested a few changes. Everything else looks good. The code is well organised.

**Comment by TejasShekar** [comment]
Implemented all PR suggestions

**Pull request merged**